src/dll/edbgEcmdDll.C

```diff
@@ -216,108 +162,59 @@ static uint32_t findChipUnitType(const ecmdChipTarget &i_target, uint64_t i_addr
-  else if (i_P9CU == PU_PHB_CHIPUNIT)     o_chipUnitType = "phb";
-  else if (i_P9CU == PU_MI_CHIPUNIT)      o_chipUnitType = "mi";
-  else if (i_P9CU == PU_DMI_CHIPUNIT)     o_chipUnitType = "dmi";
-  else if (i_P9CU == PU_MCS_CHIPUNIT)     o_chipUnitType = "mcs";
-  else if (i_P9CU == PU_MCA_CHIPUNIT)     o_chipUnitType = "mca";
-  else if (i_P9CU == PU_MCBIST_CHIPUNIT)  o_chipUnitType = "mcbist";
-  else if (i_P9CU == PU_PERV_CHIPUNIT)    o_chipUnitType = "perv";
-  else if (i_P9CU == PU_PPE_CHIPUNIT)     o_chipUnitType = "ppe";
-  else if (i_P9CU == PU_SBE_CHIPUNIT)     o_chipUnitType = "sbe";
-  else if (i_P9CU == PU_CAPP_CHIPUNIT)    o_chipUnitType = "capp";
-  else if (i_P9CU == PU_MC_CHIPUNIT)      o_chipUnitType = "mc";
-  else {
-    return out.error(EDBG_GENERAL_ERROR, FUNCNAME, "Unknown chip unit enum:%d\n", i_P9CU);
-  }
-
-  return rc;
-}
-
-//convert chipunit string to enum, as scominfo does not accept strings
-uint32_t p9n_convertCUString_to_enum(std::string cuString, p9ChipUnits_t &o_P9CU) {
-  uint32_t rc = ECMD_SUCCESS;
-
-  if (cuString == "c")          o_P9CU = PU_C_CHIPUNIT;
-  else if (cuString == "eq")    o_P9CU = PU_EQ_CHIPUNIT;
-  else if (cuString == "ex")    o_P9CU = PU_EX_CHIPUNIT;
-  else if (cuString == "xbus")  o_P9CU = PU_XBUS_CHIPUNIT;
-  else if (cuString == "obus")  o_P9CU = PU_OBUS_CHIPUNIT;
-  else if (cuString == "nv")    o_P9CU = PU_NV_CHIPUNIT;
-  else if (cuString == "pec")   o_P9CU = PU_PEC_CHIPUNIT;
-  else if (cuString == "phb")   o_P9CU = PU_PHB_CHIPUNIT;
-  else if (cuString == "mi")    o_P9CU = PU_MI_CHIPUNIT;
-  else if (cuString == "dmi")   o_P9CU = PU_DMI_CHIPUNIT;
-  else if (cuString == "mcs")   o_P9CU = PU_MCS_CHIPUNIT;
-  else if (cuString == "mca")   o_P9CU = PU_MCA_CHIPUNIT;
-  else if (cuString == "mcbist")  o_P9CU = PU_MCBIST_CHIPUNIT;
-  else if (cuString == "perv")  o_P9CU = PU_PERV_CHIPUNIT;
-  else if (cuString == "ppe")   o_P9CU = PU_PPE_CHIPUNIT;
-  else if (cuString == "sbe")   o_P9CU = PU_SBE_CHIPUNIT;
-  else if (cuString == "capp")  o_P9CU = PU_CAPP_CHIPUNIT;
-  else if (cuString == "mc")    o_P9CU = PU_MC_CHIPUNIT;
-  else {
-    return out.error(EDBG_GENERAL_ERROR, FUNCNAME, "Unknown chip unit:%S\n", cuString.c_str());
-  }
+/**
+  * @brief Get chip type
+  *
+  * @param std::string o_chipType - Chip type as output
+  *
+  * @return Upon success, ECMD_SUCCESS will be returned.  A reason code will
+  *         be returned if the execution fails.
+  */
+static uint32_t getChipType(std::string &o_chipType) {
+    int rc=ECMD_SUCCESS;
+    uint32_t l_chip_id = 0;
+    uint32_t l_address = 0x100a; //Bits 12:19 tells chipID
+    ecmdDataBuffer l_data;
+    ecmdChipTarget l_target;
+
+    l_target.cageState =  l_target.nodeState = ECMD_TARGET_FIELD_VALID;
+    l_target.slotState = l_target.posState = ECMD_TARGET_FIELD_VALID;
+
+    //Read the 0x100a cfam to get the chip id
+    rc = dllGetCfamRegister(l_target, l_address, l_data);
```

Comment:
I don't know if cfam registers can be read from the host OS to do this determination.

Reply:
Will fix it and use pdbg API's . There is an API, But all the changes are currently only in pdbg downstream. It will get to upstream in a couple of days (Delayed due to Aliaster moved out of IBM). I will use that API for now . it will break the native compilation now though but will get fixed once i update pdbg SHA to point to newer commit once the API is available

Reply:
If you make all this determination go through a common function it will be easy to swap in the pdbg API at a later date.

Reply:
Done